pkg/types/clusterconfig/network_validations.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2021 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterconfig
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/servicequotas"
+	"github.com/cortexlabs/cortex/pkg/lib/aws"
+	"github.com/cortexlabs/cortex/pkg/lib/errors"
+	"github.com/cortexlabs/cortex/pkg/lib/pointer"
+	"github.com/cortexlabs/cortex/pkg/lib/sets/strset"
+)
+
+const (
+	_elasticIPsQuotaCode         = "L-0263D0A3"
+	_internetGatewayQuotaCode    = "L-A4707A72"
+	_natGatewayQuotaCode         = "L-FE5A380F"
+	_vpcQuotaCode                = "L-F678F1CE"
+	_securityGroupsQuotaCode     = "L-E79EC296"
+	_securityGroupRulesQuotaCode = "L-0EA8095F"
+)
+
+func VerifyNetworkQuotas(
+	awsClient *aws.Client,
+	requiredInternetGateways int,
+	natGatewayRequired bool,
+	highlyAvailableNATGateway bool,
+	requiredVPCs int,
+	availabilityZones strset.Set,
+	numNodeGroups int,
+	longestCIDRWhiteList int) error {
+
+	desiredQuotaCodes := strset.New(_elasticIPsQuotaCode, _internetGatewayQuotaCode, _natGatewayQuotaCode, _vpcQuotaCode, _securityGroupsQuotaCode, _securityGroupRulesQuotaCode)
+	quotaCodeToValueMap := map[string]int{}
+
+	err := awsClient.ServiceQuotas().ListServiceQuotasPages(
```

Comment:
I think that creating a function like `GetServiceQuotas` that takes in the desired quota codes and the services to list from (`ec2` and `vpc`) that then returns a map of these might be simpler to follow. This function would then be called from here.

i.e.
```go
quotaCodeToValueMap := awsClient.GetServiceQuotas(
	[]string{
		_elasticIPsQuotaCode,
		_internetGatewayQuotaCode,
		_natGatewayQuotaCode,
		_vpcQuotaCode,
		_securityGroupsQuotaCode,
		_securityGroupRulesQuotaCode
	},
	[]string{
		"ec2",
		"vpc",
	}
)
```